### ISSUE TITLE
Shuffle fails to find next track when at last track

### ISSUE THREAD

**Issue by joostvunderink** [mazkolain-zz/spotimc#40]
First of all: nice work, mazkolain! I'm very happy to finally be able to use spotify in XBMC.

If you have a playlist, play it, go to "now playing" and enable shuffle, the playlist is then played randomly. So far, so good.

But, at some point, the last track of the playlist is played. At the end, when the next (random) song should start, instead I get the error message "Can't find a next item to play".

How to reproduce:
1) Create a playlist and play the first song
2) Go to "now playing" and make sure shuffle is enabled (you may have to disable and re-enable shuffle!)
3) Make sure that "repeat" is disabled
4) Go back to your playlist and play the last song
5) While the song is playing, click the "next" button on your remote

A side effect of this error is that shuffle is disabled, while it still seems to be on.

If you enable "repeat", the error does not appear, but playback switches from shuffle to normal, and starts over at the first track.

So, in short, shuffle works only until it happens to select the last track of the playlist, when either an error appears, or shuffle is disabled.
